evm_arithmetization/src/witness/memory.rs

```diff
@@ -200,7 +208,31 @@ impl MemoryState {
             segment,
             segment.bit_range()
         );
-        val
+        Some(val)
+    }
+
+    pub(crate) fn get_with_init(&self, address: MemoryAddress) -> U256 {
```

Comment:
nit: I got really confused when skimming through the code initially as to what this could mean. Additionally, we probably want a clear enough documentation on that method even though it's crate-restricted.